crates/rpc-types-trace/src/parity.rs

```diff
@@ -29,7 +29,7 @@ pub enum TraceType {
 #[serde(rename_all = "camelCase")]
 pub struct TraceResults {
     /// Output of the trace
-    pub output: Bytes,
```

Comment:
can we do serde(default) instead and ideally a test with a nethermind response?

Reply:
With serde(default), deserialization fails again with the same error:

> invalid type: null

We can use something like #[serde(deserialize_with = "deserialize_null_default")], but wouldn't it be better to make it optional, exactly like the vm_trace?

Reply:
hmm, I guess you're right, curious why this can be null in the first place?

Reply:
if you know why this can be null, please add it as comment

Reply:
In my opinion, this is a bug in versions prior to Nethermind/v1.26.0. Unfortunately, many EVMs use those versions.

Reply:
I see, then it would make more sense to use serde(default) imo

Reply:
I just confirmed that the bug is still not resolved in Nethermind version 1.27.0.
Using #[serde(default)] makes no difference.
#[serde(default)] will work only with workaround like here:
[https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=d9ffd2fb157f8316a2abc85e8c08dcd1](url)

So my suggestion is just making it optional.

Reply:
We can use null_as_default https://github.com/alloy-rs/alloy/blob/4ba532373eb085a5426d2de8c71c0b7273e9b62c/crates/serde/src/optional.rs#L6

Reply:
null_as_default is ok imo
I will rewrite pr accordingly

Reply:
please use serde(default)

null_as_default has a slightly different purpose